### PR TITLE
matrix return improvement

### PR DESCRIPTION
**Pull request by fabiooshiro**

### CODE REVIEW

**Comment by fabiooshiro** [comment]
@Kyle1297

File: src/Exp.js [hunk 1]
```diff
@@ -27,38 +31,29 @@ module.exports = function Exp(formula) {
             formula.cell.v = self.calc();
             formula.cell.t = getCellType(formula.cell.v);
 
-            if (Array.isArray(formula.cell.v && formula.cell.name && formula.cell.f && formula.cell.f.match(new RegExp(dynamicArrayFormulas.join('|'), 'i')))) {
+            if (isMatrix(formula.cell.v)) {
```
Comment (Kyle1297) [inline src/Exp.js:34]: Nice!

File: src/formulas.js [hunk 2]
```diff
@@ -888,4 +889,19 @@ function ceiling(number, significance) {
     return Math.ceil(number / significance) * significance
 }
 
+function filter(range, conditions) {
```
Comment (Kyle1297) [inline src/formulas.js:892]: I'm debating whether we need to have this somewhat limited FILTER function here. If you have a look at this PR here https://github.com/formulajs/formulajs/pull/119 I've created a more in-depth, tested FILTER function. Feel free to add the full logic of it here as well though (if you like).

**Review by Kyle1297 — COMMENTED** [review]
Solid 👍

File: src/formulas.js (hunk 2 above)
Comment (fabiooshiro) [reply src/formulas.js]: while formulajs guys don't merge I put your logic here on our side

**Pull request merged**